app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -34,199 +25,51 @@ import kotlinx.coroutines.withContext
 import org.mozilla.fenix.components.toolbar.BrowserInteractor
-import org.mozilla.fenix.components.toolbar.BrowserState
-import org.mozilla.fenix.components.toolbar.BrowserStore
-import org.mozilla.fenix.components.toolbar.BrowserToolbarView
-import org.mozilla.fenix.components.toolbar.DefaultBrowserToolbarController
+import org.mozilla.fenix.components.toolbar.BrowserToolbarController
 import org.mozilla.fenix.components.toolbar.QuickActionSheetAction
-import org.mozilla.fenix.components.toolbar.QuickActionSheetState
-import org.mozilla.fenix.components.toolbar.ToolbarIntegration
 import org.mozilla.fenix.customtabs.CustomTabsIntegration
-import org.mozilla.fenix.downloads.DownloadService
-import org.mozilla.fenix.ext.components
-import org.mozilla.fenix.ext.enterToImmersiveMode
 import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.ext.requireComponents
 import org.mozilla.fenix.home.sessioncontrol.SessionControlChange
 import org.mozilla.fenix.mvi.getManagedEmitter
 import org.mozilla.fenix.quickactionsheet.DefaultQuickActionSheetController
 import org.mozilla.fenix.quickactionsheet.QuickActionSheetView
-import org.mozilla.fenix.utils.Settings
 import java.net.MalformedURLException
 import java.net.URL
 
-@SuppressWarnings("TooManyFunctions", "LargeClass")
-class BrowserFragment : Fragment(), BackHandler {
-    private lateinit var browserStore: BrowserStore
-    private lateinit var browserInteractor: BrowserInteractor
-
-    private lateinit var browserToolbarView: BrowserToolbarView
+/**
+ * Fragment used for browsing the web within the main app and external apps.
+ */
+@Suppress("TooManyFunctions")
+class BrowserFragment : BaseBrowserFragment(), BackHandler {
     private lateinit var quickActionSheetView: QuickActionSheetView
 
-    private val sessionFeature = ViewBoundFeatureWrapper<SessionFeature>()
-    private val contextMenuFeature = ViewBoundFeatureWrapper<ContextMenuFeature>()
-    private val downloadsFeature = ViewBoundFeatureWrapper<DownloadsFeature>()
-    private val appLinksFeature = ViewBoundFeatureWrapper<AppLinksFeature>()
-    private val promptsFeature = ViewBoundFeatureWrapper<PromptFeature>()
-    private val findInPageIntegration = ViewBoundFeatureWrapper<FindInPageIntegration>()
-    private val toolbarIntegration = ViewBoundFeatureWrapper<ToolbarIntegration>()
     private val readerViewFeature = ViewBoundFeatureWrapper<ReaderViewFeature>()
-    private val sitePermissionsFeature = ViewBoundFeatureWrapper<SitePermissionsFeature>()
-    private val fullScreenFeature = ViewBoundFeatureWrapper<FullScreenFeature>()
     private val thumbnailsFeature = ViewBoundFeatureWrapper<ThumbnailsFeature>()
-    private val swipeRefreshFeature = ViewBoundFeatureWrapper<SwipeRefreshFeature>()
     private val customTabsIntegration = ViewBoundFeatureWrapper<CustomTabsIntegration>()
     private var findBookmarkJob: Job? = null
 
-    var customTabSessionId: String? = null
-
-    /*
```

Comment:
Since the custom tabs won't have the same type of transition should we keep this, the `view.browserLayout.transitionName = "$TAB_ITEM_TRANSITION_NAME${getSessionById()?.id}"`, `startPostponedEnterTransition()`, and the `SHARED_TRANSITION_MS` const in this file?